runtime/shared_common/CacheMap.cpp

```diff
@@ -4583,10 +4583,15 @@ SH_CacheMap::markItemStaleCheckMutex(J9VMThread* currentThread, const ShcItem* i
 	Trc_SHR_CM_markItemStaleCheckMutex_Entry(currentThread, item);
 
 	if (_ccHead->hasWriteMutex(currentThread)) {
+		if (!isCacheLocked) {
+			_ccHead->doLockCache(currentThread);		/* Wait till all readers stop and unprotect metadata area */
+			isCacheLocked = true;
+		}
 		_ccHead->markStale(currentThread, (BlockPtr)ITEMEND(item), isCacheLocked);
```

Comment:
How about using `true` instead of `isCacheLocked`, and then the parameter `isCacheLocked` doesn't need to be set to true above.

Reply:
Changed to use `true` directly.